icepyx/core/granules.py

```diff
@@ -170,14 +175,19 @@ def __init__(
     # ----------------------------------------------------------------------
     # Methods
 
-    def get_avail(self, CMRparams, reqparams, cloud=False):
+    def get_avail(
+        self,
+        CMRparams: Optional[CMRParams],
+        reqparams: Optional[EGISpecificRequiredParams],
+        cloud=False,
+    ):
         """
         Get a list of available granules for the query object's parameters.
         Generates the `avail` attribute of the granules object.
 
         Parameters
         ----------
-        CMRparams : dictionary
+        CMRparams :
```

Comment:
I find this aspect of documentation incredibly helpful (even if they have custom types defined) and get irritated when I have no idea what flavor of input I should be supplying. Is the thing to do here have a link to the particular flavor of dictionary allowed (`CMRParams`) rather than nothing?

Reply:
What I'd like to happen is for the documentation generator to pick this information up from the type hints instead of duplicating that information in the docstring. Since the docstring type information isn't actually checked, we run the risk of it becoming wrong!

We do this on _earthaccess_, but we're using mkdocs there. I have to check to make sure this is working with our doc generator.

- [ ] Check auto-generated docs include correct type information

Reply:
The auto-generated docs were not set up to parse your docstrings, which are in the numpy style. Instead, they were being rendered as straight RST. Now, the descriptions in the docstrings will be merged with the type annotations to produce a richer doc.

## New

![image](https://github.com/user-attachments/assets/c611dd46-0b6b-4965-94ff-62e8cfda0bde)


## Old

![image](https://github.com/user-attachments/assets/33176ea4-971a-462c-87f8-610ddb484361)


TODO: 

- [ ] Generate docs for the types module, ensure linkage

Reply:
> > What I'd like to happen is for the documentation generator to pick this information up from the type hints instead of duplicating that information in the docstring.
> 
> Awesome! Thanks for making that happen. How do I, as a user, now know what `CMRParamswithBbox` is? Is there a setting that will link this to the type module (so I can still know, for instance, that it's SOME type of dictionary)?

Yes, exactly. That's my intent with the TODO just above your comment. The user should be able to, once I'm done, see exactly the structure of the dict that the function expects!

>One of my documentation pet peeves is not knowing each library's custom types and having to dig into the source code to figure it out myself.

:100: Same!

Reply:
7f60849

![image](https://github.com/user-attachments/assets/fdc96bd0-b4b4-4fdf-a999-89bc630cbc9d)

There are some classes which couldn't be autodocumented because they contain symbols in the keys. There's an open issue on the sphinx repo and I linked to it in a comment!

Reply:
Thanks for all of this - the additions and the explanations. All sounds good.